src/algorithm/good_thomas_algorithm.rs

```diff
@@ -9,26 +11,22 @@ use algorithm::FFTAlgorithm;
 pub struct GoodThomasAlgorithm<T> {
     width: usize,
     // width_inverse: usize,
-    width_size_fft: Box<FFTAlgorithm<T>>,
+    width_size_fft: Rc<FFTAlgorithm<T>>,
 
     height: usize,
     // height_inverse: usize,
-    height_size_fft: Box<FFTAlgorithm<T>>,
+    height_size_fft: Rc<FFTAlgorithm<T>>,
```

Comment:
I've been thinking about this exact issue: The library's design currently isn't great for a memory-constrained system, and doesn't work at all if you want zero allocations.

What you're proposing is a neat idea, and it will work for some situations. For example, i think it would be a big improvement to handle Raders' Algorithm this way. But if we're disallowing trait objects (which require heap allocations) then the type of the "parent" must be known at compile time, which would imply that the size of the FFT we want to compute is known at compile time.

There is an alternative to trait objects though: enums. We could create an enum containing every FFT algorithm, and then write a recursive mixed radix implementation that contains a Vec of those enums. That would still require a heap allocation for the Vec, though, but at least we could avoid Box and Rc.

If you want to avoid heap allocations altogether, I've actually been thinking about making another FFT crate called TinyFFT that eliminates heap allocations. It wouldn't compete with RustFFT on performance, but for memory-constrained environments or for no-std environments it would be a useful alternative.